includes/class-amp-story-post-type.php

```diff
@@ -1115,4 +1118,110 @@ public static function get_featured_image_minimum_height( $posts ) {
 
 		return $minimum_height;
 	}
+
+	/**
+	 * Crops the image and returns the object as JSON.
+	 *
+	 * Forked from Custom_Image_Header::ajax_header_crop().
+	 */
+	public static function crop_featured_image() {
+		check_ajax_referer( 'image_editor-' . $_POST['id'], 'nonce' );
+
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			wp_send_json_error();
+		}
+
+		$crop_details = $_POST['cropDetails'];
+
+		$dimensions = array(
+			'dst_width'  => 696,
+			'dst_height' => 928,
+		);
+
+		$attachment_id = absint( $_POST['id'] );
+
+		$cropped = wp_crop_image(
+			$attachment_id,
+			intval( $crop_details['x1'] ),
+			intval( $crop_details['y1'] ),
+			intval( $crop_details['width'] ),
+			intval( $crop_details['height'] ),
+			intval( $dimensions['dst_width'] ),
+			intval( $dimensions['dst_height'] )
+		);
+
+		if ( ! $cropped || is_wp_error( $cropped ) ) {
+			wp_send_json_error( array( 'message' => __( 'Image could not be processed. Please go back and try again.', 'default' ) ) );
+		}
+
+		/** This filter is documented in wp-admin/custom-header.php */
+		$cropped = apply_filters( 'wp_create_file_in_uploads', $cropped, $attachment_id ); // For replication.
+		$object  = self::create_attachment_object( $cropped, $attachment_id );
+		unset( $object['ID'] );
+
+		$new_attachment_id       = self::insert_attachment( $object, $cropped );
+		$object['attachment_id'] = $new_attachment_id;
+		$object['url']           = wp_get_attachment_url( $new_attachment_id );
+		$object['width']         = $dimensions['dst_width'];
+		$object['height']        = $dimensions['dst_height'];
+
+		wp_send_json_success( $object );
+	}
+
+	/**
+	 * Create an attachment 'object'.
+	 *
+	 * Forked from Custom_Image_Header::create_attachment_object() in Core.
+	 *
+	 * @param string $cropped Cropped image URL.
+	 * @param int    $parent_attachment_id Attachment ID of parent image.
+	 * @return array Attachment object.
+	 */
+	public static function create_attachment_object( $cropped, $parent_attachment_id ) {
+		$parent     = get_post( $parent_attachment_id );
+		$parent_url = wp_get_attachment_url( $parent->ID );
+		$url        = str_replace( basename( $parent_url ), basename( $cropped ), $parent_url );
+		try {
+			$size = getimagesize( $cropped );
+		} catch ( Exception $error ) {
+			unset( $error );
+		}
+
+		$image_type = ( $size ) ? $size['mime'] : 'image/jpeg';
+		$object     = array(
+			'ID'             => $parent_attachment_id,
+			'post_title'     => basename( $cropped ),
+			'post_mime_type' => $image_type,
+			'guid'           => $url,
+			'context'        => 'custom-header',
```

Comment:
Can we change this to `amp-story-poster` or something? Where is this used?

Reply:
Sure, https://github.com/ampproject/amp-wp/pull/2031/commits/d2d69981d3464e764f5c976013af7ebcb0c11537 changes the context to `amp-story-poster`. 

I'm not sure where it's used. The only thing I could find in [customize-controls.js](https://github.com/WordPress/wordpress-develop/blob/c80325658f85d24ff82295dd2d55bfdf789f4163/src/js/_enqueues/wp/customize/controls.js#L4742) that might relate to this is:

```js
wp.ajax.post( 'crop-image', {
	nonce: attachment.nonces.edit,
	id: attachment.id,
	context: 'site-icon',
````

But that's sending a POST request, not responding to one.